### DISCUSSION TITLE
프론트 추가 요구 사항

### DISCUSSION THREAD

**Discussion by jaewonLeeKOR** [UMC-TEAM-RF/RF-Server#158] [General]
# 1. 유저별 소켓 연결 필요

- 가입 신청 알림, 가입수락 여부 알림을 앱 내에서 실시간으로 표시하기 위함

# 2. 로그인 api 수정

- 로그인 시 유저의 알프 온도 데이터 추가

# 3. 유저 조회 API 수정

- 사용자 조회 시 추가 데이터 요청
  - 알프 온도
  - 사용자 정보 조회를 요청한 주체 사용자가 목표 사용자를 평가 했는지 여부
    - 좋아요를 했는지, 싫어요를 했는지 포함

# 4. 회원가입 API 수정

- 회원가입 시 소속학과도 같이 입력하도록 되어 있는지 확인 요망
- 해당 사항 미구현 상태면 구현 요망

# 5. 아이디 찾기, 비밀번호 재설정 API 구현

# 6. 로그아웃 API 구현

# 7. 유저 차단 및 해제 API 구현

# 8. 알림 리스트 API

# 9. 가입 크루 리스트 전체 조회 API

- 채팅에 필요한 채팅방 리스트를 불러오기 위함
- response
  ```json
  {
    "partyId" : {partyId},
    "partyName" : {partyName}
  }
  ```
- 같이 주면 좋은 데이터
  - 모임에 가입된 유저 수
  - 모임 배너 이미지 url

# 10. 일정 불러오기 HTTP method 수정 요청

- get -> post 수정 필요

# 11. 이미지 데이터 URL 변환 API 구현

# 원문

https://www.notion.so/j-0ne/6e9cffcadf6c4296a733aebe5e5ea3e7?pvs=4

**Comment by jaewonLeeKOR** [comment]
@kuk6933 @minjgziii @lxxyxin 
확인하시고 각 부분에 대한 의견을 주세요.
필요없을거 같다거나, 이미 구현된 부분이라거나, 진행하고 있으면 진행사항 알려주세요~!

**Comment by jaewonLeeKOR** [comment]
### 1번 요구사항
- 기존에 구현된 푸시 알림을 통해 대체할 수 있는 작업인거 같아요.
- 해당 기능으로 대체될 수 있는지 알아보겠습니다.

### 8번 요구사항
- 현재 작업중이며 테스트 단계에 있습니다.

### 9번 요구사항
- 해당 API 는 로그인시 채팅방의 연결을 위해 사용하는 기능이므로 다른 데이터를 더 받는것은 좋지 않은거 같아요.
- 모입에 가입한 유저 수와 모임 배너 이미지url은 따로 API 를 사용해서 조회를 하는것이 좋을것 같습니다.

**Comment by yxinot** [comment]
10번은 왜 수정해야하는지 궁금합니다. 조회 요청은 GET으로 해야하는것으로 아는데 일정을 생성하는게 아님에도 불구하고 POST로 바꿔야 하는 이유가 있나요?

**Reply by jaewonLeeKOR** [reply]
https://www.notion.so/j-0ne/7b76d07679ae4c089c03ff1cd00a57a1
이 부분에서 이전에 request body를 API 시트에 작성한 이력이 있어서 그런가보네요. 원문에 추가적으로 작성된 내용을 보면 이전에 프론트 측에서 post 방식으로 변경하겠다는 답변을 받았다고 하는데 맞나요?
더해 수정중이라고 작성되어있는데 수정이 완료가 된건가요?
@lxxyxin

**Reply by yxinot** [reply]
디스코드 보시면 query string으로 바꾼다고 했고 POST로 바꾸겠다는 답변은 드린적이 없습니다. 아무래도 저때 소통에 약간 오류가 있었나보네요... ㅎㅎ 수정은 이미 완료된 상태고 머지만 하면 됩니다!

**Comment by kuk6933** [comment]
### 4번 요구사항
- 구현되어 있습니다. 다만 user정보 받아오는데에는 누락되어 있어서 추가하겠습니다

### 6번 요구사항
- 현재 작업중입니다

### 9번 요구사항
- 구현 하겠습니다

**Comment by minjgziii** [comment]
### 5번 요구사항
- 아이디 찾기 구현 완료됐었는데 아이디 찾기할 때 닉네임 빼는 거 전 전달을 받은 적이 없어서.. 다시 작업하겠습니다. 바뀌면 저한테도 알려주세요....ㅜ
- 비밀번호 재설정은 아이디 찾기 이후 작업 예정입니다.

### 7번 요구사항
- 추후 작업 예정입니다.
